File: plugins/src/main/plugin/buttons/ButtonClick.java
package plugin.buttons;

import astraeus.game.event.EventContext;
import astraeus.game.event.EventSubscriber;
import astraeus.game.event.impl.ButtonActionEvent;
import astraeus.game.model.entity.mob.player.Player;

public abstract class ButtonClick implements EventSubscriber<ButtonActionEvent> {

	@Override
	public void subscribe(EventContext context, Player player, ButtonActionEvent event) {
		execute(player, event);
	}
	
	protected abstract void execute(Player player, ButtonActionEvent event);
	
	public abstract boolean test(ButtonActionEvent event);

}

File: plugins/plugin/buttons/ButtonClick.kt
package plugin.buttons

import astraeus.game.event.EventContext
import astraeus.game.event.EventSubscriber
import astraeus.game.event.impl.ButtonActionEvent
import astraeus.game.model.entity.mob.player.Player

public abstract class ButtonClick : EventSubscriber<ButtonActionEvent> {

	override fun subscribe(context: EventContext, player: Player, event: ButtonActionEvent) {
		execute(player, event)
	}

	abstract fun execute(player: Player, event: ButtonActionEvent)
}